### PR TITLE
Fix error when reimporting an Image resource

### LINKED ISSUE #21276: Reimport a png file as Image it shows an error
**Godot version:**
v3.1.dev.calinou.8bc9986

**OS/device including version:**
windows 10 x64

**Issue description:**
when Reimport a png file as Image it shows an error :

ERROR: get_section_keys: Condition ' !values.has(p_section) ' is true.
   At: core/io/config_file.cpp:116

**Step to produce:**
This occurs when change the default icon.png
Click on the default Icon.png on "file system" tab (or any image file) >> go to Import tab >> change "import as" to Image >> click reimport .

### PR DESCRIPTION
**Pull request by YeldhamDev**
Fixes #21276.

### CODE REVIEW

File: editor/editor_file_system.cpp [hunk 1]
```diff
@@ -1478,7 +1478,7 @@ void EditorFileSystem::_reimport_file(const String &p_file) {
 		Ref<ConfigFile> cf;
 		cf.instance();
 		Error err = cf->load(p_file + ".import");
-		if (err == OK) {
+		if (err == OK && cf->has_section("params")) {
```
Comment (akien-mga) [inline editor/editor_file_system.cpp]: I haven't looked into details, but isn't the `importer_name = cf->get_value("remap", "importer");` call in the `if` still relevant even if the `params` section is missing?

Comment (YeldhamDev) [reply editor/editor_file_system.cpp]: Yep, just caught that, making the necessary changes.

**Comment by YeldhamDev** [comment]
Good to go.

File: editor/editor_file_system.cpp [hunk 2]
```diff
@@ -1479,10 +1479,12 @@ void EditorFileSystem::_reimport_file(const String &p_file) {
...
-			for (List<String>::Element *E = sk.front(); E; E = E->next()) {
-				params[E->get()] = cf->get_value("params", E->get());
+			if (cf->has_section("params")) {
+				List<String> sk;
+				cf->get_section_keys("params", &sk);
+				for (List<String>::Element *E = sk.front(); E; E = E->next()) {
+					params[E->get()] = cf->get_value("params", E->get());
+				}
 			}
 			importer_name = cf->get_value("remap", "importer");
```
Comment (akien-mga) [inline editor/editor_file_system.cpp]: Maybe this also needs a `if (cf->has_section("remap"))`?

**Pull request merged**